includes/ucf-spotlight-common.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/**
+ * Place common functions here.
+ **/
+
+if ( ! class_exists( 'UCF_Spotlight_Common' ) ) {
+	class UCF_Spotlight_Common {
+
+		public function enqueue_styles() {
+			if ( get_option( 'ucf_spotlight_include_css' ) ) {
+				wp_enqueue_style( 'ucf_spotlight_css', plugins_url( 'static/css/ucf-spotlight.min.css', UCF_SPOTLIGHT__PLUGIN_FILE ), false, false, 'all' );
+			}
+		}
+
+		public static function display_spotlight( $attr ) {
+			$output = ucf_spotlight_display( $attr );
+			return apply_filters( 'ucf_spotlight_display', $output );
+		}
+	}
+
+	add_action( 'wp_enqueue_scripts', array( 'UCF_Spotlight_Common', 'enqueue_styles' ), 99 );
+}
+
+/**
+* Returns the spotlight HTML to be displayed on the page.
+* @author RJ Bruneel
+* @since 1.0.0
+* @param $attr string | slug of the spotlight post type and one of three layouts.
+* @return String
+**/
+if ( ! function_exists( 'ucf_spotlight_display' ) ) {
+	function ucf_spotlight_display( $attr ) {
+		// get post with $slug
+		if( !empty( $attr['slug'] ) ) {
+
+			$args = array(
+				'name'        => $attr['slug'],
+				'post_type'   => 'ucf_spotlight',
+				'post_status' => 'publish',
+				'numberposts' => 1
+			);
+			$post = array_shift( get_posts( $args ) );
+
+			$spotlight_image = ( $thumb = get_post_thumbnail_id( $post->ID ) ) ? array_shift( wp_get_attachment_image_src( $thumb, 'single-post-thumbnail' ) ) : NULL;
+
+			$args = array(
+				'layout'    => get_post_meta( $post->ID, "ucf_spotlight_layout", True ),
+				'header'    => get_post_meta( $post->ID, "ucf_spotlight_header", True ),
+				'copy'      => get_post_meta( $post->ID, "ucf_spotlight_copy", True ),
+				'link_text' => get_post_meta( $post->ID, "ucf_spotlight_link_text", True ),
+				'link_url'  => get_post_meta( $post->ID, "ucf_spotlight_link_url", True ),
+				'image'     => $spotlight_image,
+			);
+
+			switch ( $args['layout'] ) {
+				case "horizontal":
+					return ucf_spotlight_horizontal( $args );
+				case "vertical":
+					return ucf_spotlight_vertical( $args );
+				case "square":
+					return ucf_spotlight_square( $args );
+				default:
+					return ucf_spotlight_square( $args );
+			}
+		}
+	}
+}
+
+/**
+* Returns the spotlight HTML for the horizontal layout of the spotlight.
+* @author RJ Bruneel
+* @since 1.0.0
+* @param $attr string | contains the various elements of the spotlight (header, copy, link text and link url).
+* @return String
+**/
+if ( ! function_exists( 'ucf_spotlight_horizontal' ) ) {
+	function ucf_spotlight_horizontal( $args ) {
+		ob_start();
+	?>
+				</div>
+			</div>
+		</div>
+		<section class="spotlight-horizontal" style="background-image: url(<?php echo $args['image'] ?>)">
```

Comment:
Consider creating and pulling multiple image sizes (using custom thumbnails) to use at different screen widths to avoid loading large background images on mobile devices--or, remove background images entirely at mobile sizes, so we don't even have to think about making background images scale down for these.